packages/runtime/container-runtime/src/gc/garbageCollection.ts

```diff
@@ -537,30 +664,145 @@ export class GarbageCollector implements IGarbageCollector {
 			this.getLastSummaryTimestampMs(),
 		);
 
-		let updatedGCData: IGarbageCollectionData = gcData;
-
-		if (this.configs.shouldRunSweep) {
-			updatedGCData = this.runSweepPhase(sweepReadyNodes, gcData);
-		} else if (this.configs.testMode) {
-			// If we are running in GC test mode, delete objects for unused routes. This enables testing scenarios
-			// involving access to deleted data.
+		/**
+		 * Currently, there are 3 modes for sweep:
+		 * Test mode - Unreferenced nodes are immediately deleted without waiting for them to be sweep ready.
+		 * Tombstone mode - Sweep ready modes are marked as tombstones instead of being deleted.
+		 * Sweep mode - Sweep ready modes are deleted.
+		 *
+		 * These modes serve as staging for applications that want to enable sweep by providing an incremental
+		 * way to test and validate sweep works as expected.
+		 */
+		if (this.configs.testMode) {
+			// If we are running in GC test mode, unreferenced nodes (gcResult.deletedNodeIds) are deleted.
 			this.runtime.updateUnusedRoutes(gcResult.deletedNodeIds);
-		} else if (this.configs.tombstoneMode) {
+			return gcData;
```

Comment:
Do we not need to update gcData here?

Reply:
Not with how test mode currently works. I am assuming you mean updating gcData similar to how it happens with actual sweep.